security/advancedtls/advancedtls.go

```diff
@@ -475,3 +478,28 @@ func cloneTLSConfig(cfg *tls.Config) *tls.Config {
 	}
 	return cfg.Clone()
 }
+
+// GetCertificateWithSNI is a helper function that returns the certificate
+// that matches the SNI field based on the given ClientHelloInfo
+// if GetCertificate returns a list of certificates
+func (o *ServerOptions) getCertificateWithSNI(clientHello *tls.ClientHelloInfo) (*tls.Certificate, error) {
```

Comment:
Let's make the function signature to be 
```
buildGetCertificateFunc(clientHello *tls.ClientHelloInfo, o *ServerOptions) func(clientHello *tls.ClientHelloInfo) (*tls.Certificate, error)
```
So basically we are trying to make it clear that here we are just trying to build the function instead of actually executing them.

Reply:
I see, that makes sense!